File: cpp/tests/ConvertSortedArrayToBinarySearchTreeTest.cpp
#include "ConvertSortedArrayToBinarySearchTree.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(ConvertSortedArrayToBinarySearchTree, Solution108_1) {
  auto S108_1 = Solution108_1();

  std::vector<int> Nums1{-10, -3, 0, 5, 9};
  auto Expected1 = newTree({0, -3, 9, -10, NTNode, 5});
  auto Actual1 = S108_1.sortedArrayToBST(Nums1);
  EXPECT_EQ(*Expected1, *Actual1);
  deleteTree(Expected1);
  deleteTree(Actual1);

  std::vector<int> Nums2{1, 3};
  auto Expected2 = newTree({3, 1});
  auto Actual2 = S108_1.sortedArrayToBST(Nums2);
  EXPECT_EQ(*Expected2, *Actual2);
  deleteTree(Expected2);
  deleteTree(Actual2);
}

File: java/src/test/java/io/binac/leetcode/ConvertSortedArrayToBinarySearchTreeTest.java
package io.binac.leetcode;

import io.binac.leetcode.util.TreeNode;
import io.binac.leetcode.util.Trees;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ConvertSortedArrayToBinarySearchTreeTest {
    private static final ConvertSortedArrayToBinarySearchTree.Solution1 SOLUTION1 = new ConvertSortedArrayToBinarySearchTree.Solution1();

    private final int[] NUMS1 = {-10, -3, 0, 5, 9};
    private final TreeNode EXPECTED1 = Trees.asTree(0, -3, 9, -10, null, 5);

    private final int[] NUMS2 = {1, 3};
    private final TreeNode EXPECTED2 = Trees.asTree(3, 1);


    @Test
    void test1() {
        assertTrue(Trees.equals(EXPECTED1, SOLUTION1.sortedArrayToBST(NUMS1)));
        assertTrue(Trees.equals(EXPECTED2, SOLUTION1.sortedArrayToBST(NUMS2)));
    }
}